### DISCUSSION TITLE
N-HiTS for Classification

### DISCUSSION THREAD

**Discussion by dero03** [Nixtla/neuralforecast#336] [Q&A] [answered]
I was wondering if nhits could be used not only for regression but also for classification (maybe adding a softmax layer). Is it possible or is a stupid idea?

**Comment by kdgutier** [comment, marked as answer]
Hi @Bl4ckD3ro,

It seems a good idea, and we might not be so far away from implementing it.
I am currently working to include PyTorch distribution outputs to the neuralforecast models, it is work in progress.
You might want to take a look here: https://github.com/Nixtla/neuralforecast/blob/DeepAR/nbs/losses.pytorch.ipynb

**Reply by kdgutier** [reply]
I also think that something similar was done in the M5 competition data, where they adapted the Zero inflated regression loss (Dirac+Poisson mixture) with an N-BEATS. For the moment I don't have links to the implementation but it is related to the topic.
